### PR TITLE
JWT policy tests 

### PR DESCRIPTION
**Pull request by vepatel**
### Proposed changes
- Add tests for JWT policy (VirtualServer),
- Minor refactoring and updates to common utils.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: tests/suite/custom_resources_utils.py [hunk 1]
```diff
@@ -329,14 +329,15 @@ def patch_virtual_server_from_yaml(
...
-        custom_objects.patch_namespaced_custom_object(
-            "k8s.nginx.org", "v1", namespace, "virtualservers", name, dep
+    try: 
+        delete_virtual_server(
+            custom_objects, name, namespace
+        )
+        create_virtual_server_from_yaml(
+            custom_objects, yaml_manifest, namespace
         )
-        print(f"VirtualServer updated with name '{dep['metadata']['name']}'")
```
Comment (Dean-Coakley) [inline tests/suite/custom_resources_utils.py:339]: was this removal intentional?

**Review by Dean-Coakley — APPROVED** [review]

File: tests/suite/custom_resources_utils.py (hunk 1 above)
Comment (vepatel) [reply tests/suite/custom_resources_utils.py]: Fixed in https://github.com/nginxinc/kubernetes-ingress/pull/1230/commits/d268b7c2fb71e4037891035e2c2ac867ef9a419c

**Review by LorcanMcVeigh — APPROVED** [review]
